torch/csrc/jit/netdef_converter.cpp

```diff
@@ -108,6 +112,31 @@ void convertNetDefToIR(
   for (const auto& outputName : net.external_output()) {
     AT_ASSERT(valueMap.count(outputName));
     g->registerOutput(valueMap.at(outputName));
+    namesMap[valueMap.at(outputName)] = outputName;
+  }
+
+  // Set proper unique names for all values.
+  // We will set the names for external inputs and outputs last, so that if the
+  // names are reused, then intermediate values will be renamed and the external
+  // values will keep the original names.
+  for (const auto& kv : namesMap) {
```

Comment:
This is going to result in non-deterministic names because the order of iteration is not specified. It would be better to visit the values in order and do naming in that order (or some other purely deterministic scheme).

Reply:
Good catch, I'll fix that!